app/usecases/users.py

```diff
@@ -192,3 +200,142 @@ async def update_email_address(
+    # - [delete; key'd by username??] password_recovery
+    # - [TODO/AC] patcher_detections
+    # - [TODO/AC] patcher_token_logs
+    # - [TODO] profile_backgrounds (and filesystem data)
+    # - [TODO] rap_logs
+    # - [leave as-is] remember
+    # - [leave as-is] reports
+    # - [leave as-is] rework_queue
+    # - [leave as-is] rework_scores
+    # - [leave as-is] rework_stats
+    # - [leave as-is] scheduled_bans
+    # - [leave as-is] scores
+    # - [leave as-is] scores_ap
+    # - [leave as-is] scores_relax
+    # - [leave as-is] scores_first
+    # - [TODO/AC] score_submission_logs
+    # - [leave as-is] tokens
+    # - [leave as-is] user_relationships
+    # - [leave as-is] user_beatmaps
+    # - [leave as-is] user_favourites
+    # - [leave as-is] user_profile_history
+    # - [leave as-is] user_speedruns
+    # - [leave as-is] user_tokens
+
+    # misc.
+    # - [anonymize] replay data for all scores
+    # - [TODO] youtube uploads
+
+    # PII to focus on:
+    # - username / username aka
+    # - email
+    # - clan association
+    # - country
+    # - (potetnailly) user notes
+    # - (potentially) userpage content
+
+    transaction = await app.state.database.transaction()
+    try:
+        user = await users.fetch_one_by_user_id(user_id)
+        if user is None:
+            return Error(
+                error_code=ErrorCode.NOT_FOUND,
+                user_feedback="User not found.",
+            )
+
+        if user.clan_id:
+            clan = await clans.fetch_one_by_clan_id(user.clan_id)
+            if clan is not None:
+                if user.id == clan.owner:
+                    # transfer clan ownership to another member, if available
+                    other_clan_members = sorted(
+                        [
+                            u
+                            for u in await users.fetch_many_by_clan_id(user.clan_id)
+                            if u.id != user.id
+                        ],
+                        # XXX: heuristic; clan join date would be better
+                        #      but it is not something we currently store
+                        key=lambda u: (u.privileges, u.latest_activity),
+                    )
+                    if other_clan_members:
+                        new_owner = other_clan_members[0]
+                        await clans.update_owner(user.clan_id, new_owner.id)
+                    else:
+                        # no other members in the clan; just delete it
+                        await clans.delete_one_by_clan_id(user.clan_id)
+
+        await password_recovery.delete_many_by_username(user.username)
+
+        # TODO: consider what ac data should be anonymized instead of wiped
+        await user_ip_associations.delete_many_by_user_id(user_id)
+        await user_hwid_associations.delete_many_by_user_id(user_id)
+        await lastfm_flags.delete_many_by_user_id(user_id)
+        # TODO: patcher_detections & patcher_token_logs
```

Comment:
@tsunyoku do you know how can i identify records from a given user in these?

Reply:
both tables have a `score_id` column you can do a join by. you'll also want to clear `score_submission_logs`, which also has a `score_id` column

Reply:
I see `patcher_token_logs` has a score id, ty - but `patcher_detections` doesnt seem to. Do you know what I could use?

Reply:
Going to merge for now and revisit this